### PR TITLE
Add thruster tool

### PR DESCRIPTION
**Pull request by ninjrdevelop**
Is currently use-activated. Needs an effect added to it.

### CODE REVIEW

File: Code/Components/Tools/Thruster.cs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+	protected override void OnUpdate()
+	{
+		base.OnUpdate();
+		if ( IsProxy )
+			return;
+
+		if ( timeSinceDisabled < Time.Delta * 5f || !Parent.IsValid() )
+			return;
+	}
+
```
Comment (Nebual) [inline Code/Components/Tools/Thruster.cs:56]: 🌵 This function has no effects
```suggestion
```

File: Code/Components/Tools/Thruster.cs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+	public override bool Primary( SceneTraceResult trace )
+	{
+		if ( !trace.Hit || !trace.GameObject.IsValid() )
+			return false;
+
+		if ( Input.Pressed( "attack1" ) )
+		{
+			if ( !trace.Hit || !trace.GameObject.IsValid() || trace.Tags.Contains( "player" ) ) return false;
+			if ( trace.Tags.Contains( this.tagName() ) )
+				return true;
```
Comment (Nebual) [inline Code/Components/Tools/Thruster.cs:66]: This chunk conflicts with L68-74 (the update-in-place) - you won't find a ThrusterComponent if you return early when finding the tag
```suggestion
```

File: Code/Components/Tools/Thruster.cs [hunk 3]
```diff
@@ -0,0 +1,197 @@
+﻿using Sandbox.UI;
+
+[Library( "tool_thruster", Description = "A rocket type thing that can push forwards and backward", Group = "construction" )]
+public class Thruster : BaseTool
+{
+	RealTimeSince timeSinceDisabled;
```
Comment (Nebual) [inline Code/Components/Tools/Thruster.cs:6]: 🌵 I don't think this is doing anything (here nor in the Wheel tool lol)

File: Code/Components/ToolCreations/ThrusterComponent.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+		}
+	}
+
+	protected void OnThrusterEnabled()
+	{
+		// Turn emitter on
+		if ( effects != null )
+			return;
+
+		effects = Particles.MakeParticleSystem( "particles/physgun_end_nohit.vpcf", Transform.World, 0, GameObject );
```
Comment (Nebual) [inline Code/Components/ToolCreations/ThrusterComponent.cs:38]: I ported over the old effect (which isn't ideal, I want a flame haha)

Be sure to `git pull` to grab my commit before editing this branch :)

**Review by Nebual — COMMENTED** [review]
Thanks for getting that working! Thrusterrrsssss

File: Code/Components/Tools/Thruster.cs (hunk 3 above)
Comment (ninjrdevelop) [reply Code/Components/Tools/Thruster.cs:6]: Do you mean the RealTimeSince?

File: Code/Components/ToolCreations/ThrusterComponent.cs (hunk 4 above)
Comment (ninjrdevelop) [reply Code/Components/ToolCreations/ThrusterComponent.cs:38]: Nice!

File: Code/Components/Tools/Thruster.cs (hunk 3 above)
Comment (Nebual) [reply Code/Components/Tools/Thruster.cs:6]: Yeah, nbd though

**Review by Nebual — APPROVED** [review]

**Pull request merged**